charts/externalcreds/templates/deployment.yaml

```diff
@@ -1,11 +1,13 @@
-{{- $imageTag := .Values.imageConfig.tag | default .Values.global.applicationVersion -}}
+{{- $imageTag := $.Values.imageConfig.tag | default $.Values.global.applicationVersion -}}
+{{- range $.Values.appDeployments }}
```

Comment:
This `range` function will create the included block for each `appDeployments` in values.yaml. Most of the changes below are changing `.` (current scope) to `$` (root scope) because `range` creates a new scope so the current scope is no longer the root scope as it was before.